src/mpid/ch4/generic/am/mpidig_am_part_callbacks.c

```diff
@@ -152,7 +152,7 @@ int MPIDIG_part_send_data_target_msg_cb(int handler_id, void *am_hdr, void *data
     MPIDIG_REQUEST(rreq, req->target_cmpl_cb) = part_send_data_target_cmpl_cb;
 
     /* Set part_rreq complete when am request completes but not decrease part_rreq refcnt */
-    rreq->completion_notification = &part_rreq->cc;
+    rreq->cc_ptr = &part_rreq->cc;
```

Comment:
If there are more than 1 sub-request, `part_rreq->cc` could be more than `1`, and setting `cc_ptr` to parent `cc` won't be able to complete the sub-request itself. 

While this works right now, I think it puts a obscure restriction.

Reply:
Yes, that's true. I mainly did this as an exercise to fully understand the completion mechanism. We still may want to modify it to use multiple sends, say for the POSIX shm path. So we can just close this.